### ISSUE TITLE
Support two or more multidatacenter on same nodes?

### ISSUE THREAD

**Issue by joseluissmarting** [k8ssandra/k8ssandra-operator#1443]
**What is missing?**
Support two or more multidatacenter on same nodes: 
- https://docs.k8ssandra.io/install/local/multi-cluster-helm/

**Why do we need it?**
We use a lot of tables and differents apps in the same cassandra cluster multidatacenter. We need to separate this cluster into several, to avoid the impact in Cassandra: 
- https://docs.datastax.com/en/planning/oss/data-model.html#number-of-tables

**Environment**
Cassandra 5.0.1

* K8ssandra Operator version:
cr.k8ssandra.io/k8ssandra/k8ssandra-operator:v1.20.2

    
    

**Anything else we need to know?**:
You already support the change of the hostPort: https://docs-v2.k8ssandra.io/reference/crd/k8ssandra-operator-crds-latest/#k8ssandraclusterspeccassandracontainersindexportsindex

But we can't change the default ports used in the cointainers images and it gives us erros like:
` ERROR [main] 2024-10-29 10:38:27,940 CassandraDaemon.java:887 - Port already in use: 7199; nested exception is:
java.net.BindException: Address already in use (Bind failed) `

I only see this one: MGMT_API_LISTEN_TCP_PORT: 8080 , https://github.com/k8ssandra/management-api-for-apache-cassandra/issues/398

Thanks you in advance!

**Comment by burmanm** [comment]
We already support running multiple Cassandra pods on the same nodes. It's just affinities. What you're seeing is probably the use of hostNetworking which then takes the ports on the host. If using pod network, this isn't an issue, you can have as many pods as you want with the default port settings.

**Comment by joseluissmarting** [comment]
Hi burmanm,
thanks for the response.
Yes, I know that, but I need the hostNetwork for the multi datacenter comunication. It uses the default ports for the nodes and it's not configurable, the ports o nodes. You only specify a k8scontext.

**Comment by burmanm** [comment]
Yes, we don't support port changes at this point from the CRD. For mgmt-api configuration, it would be this ticket: https://github.com/k8ssandra/cass-operator/issues/242

In theory, the PodTemplateSpec in CassandraDatacenter allows to change the ports, but I'm not sure if the information is shared correctly (as only the mgmt-api port is stored in the status that's used for connection from operator <-> mgmt-api).
